### PR TITLE
ZMP problem update

### PR DESCRIPTION
**Pull request by iamsavva**
major changes are:

- more detailed system description
- add part (a) -- asking for free body diagrams
- add part (e) -- what does it mean to be falling down?
- part (h) rephrased to clarify what "designing a robot that walks forward" means = "easy to lift up heel"

---

including the new problem below for reference:

Screenshot 2024-04-12 at 1 48 09 PM

- - -
This change is [Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/556)

### CODE REVIEW

**Comment by iamsavva** [comment]
CI checks are failing, but it has nothing to do with the changes.

- macOS Sonoma 14 fails because  `/book/htmlbook/workflows/ci/` has setup files for  `focal`, `jammy`, `monterey`, but not `sonoma`.
- the other one fails because it can't find the `planar_one_leg_hopper.dmd.yaml`

**Review by jaehunchoe827 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/556)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @iamsavva)

___
*[`book/humanoids.html` line 558 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/556#-NvE3vHlDsXzrxWDOT_Z:-NvE3vHlDsXzrxWDOT__:bixaonc) ([raw file](https://github.com/RussTedrake/underactuated/blob/f8bbb54ff65e5d3cbec1bcfca2384a43633d869a/book/humanoids.html#L558)):*
> ```HTML
>       </li>
> 
>       <li>cd 
> ```

typo: Could you remove this ('cd')?

**Review by iamsavva — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/556)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @jaehunchoe827)

___
*[`book/humanoids.html` line 558 at r1](https://reviewable.io/reviews/RussTedrake/underactuated/556#-NvE3vHlDsXzrxWDOT_Z:-NvE6s4Y0j1r5xY-wnjk:b-7ql1td) ([raw file](https://github.com/RussTedrake/underactuated/blob/f8bbb54ff65e5d3cbec1bcfca2384a43633d869a/book/humanoids.html#L558)):*
<details><summary><i>Previously, jaehunchoe827 wrote…</i></summary><blockquote>

typo: Could you remove this ('cd')?
</blockquote></details>

fixed

**Comment by RussTedrake** [comment]
Agreed that ci is not due to this pr

File: book/humanoids.html [hunk 1]
```diff
@@ -508,65 +508,85 @@ <h1><a href="index.html" style="text-decoration:none;">Underactuated Robotics</a
...
-        <figcaption>External forces acting on a robot pushing on a flat ground</figcaption>
+        <img width="50%" src="figures/exercises/foot_on_ground.svg"/>
+        <figcaption>Model of a simplified legged robot on flat ground.</figcaption>
       </figure>
-      Consider an abstracted legged robot. The foot and the leg are assumed to be massless. Suppose there is no actuator at the pin joint P
-     <ol type = "a">
-       <li> 
+      Consider an abstracted legged robot.
+      The foot (enclosed by points HTP) and the leg of length $l$ are assumed to be massless.
+      The bob at the end of the leg has a point mass $m$ and represents the robot's body.
```
[the remark this answers is not in the document]
Comment (iamsavva) [reply book/humanoids.html]: fixed

File: book/robust.html [hunk 2]
```diff
@@ -724,6 +724,12 @@ <h1><a href="index.html" style="text-decoration:none;">Underactuated Robotics</a
 PhD thesis, Massachusetts Institute of Technology, September, <span class="year">2014</span>.
 [&nbsp;<a href="http://groups.csail.mit.edu/robotics-center/public_papers/Moore14b.pdf">link</a>&nbsp;]
 
+</li><br>
+<li id=Boyd04a>
+<span class="author">Stephen Boyd and Lieven Vandenberghe</span>, 
+<span class="title">"Convex Optimization"</span>, Cambridge University Press
+, <span class="year">2004</span>.
```
[the remark this answers is not in the document]
Comment (iamsavva) [reply book/robust.html]: my mistake -- I used an outdated version of `install_html_meta_data.py` script. Fixed.

File: book/humanoids.html [hunk 3]
```diff
@@ -508,65 +508,85 @@ <h1><a href="index.html" style="text-decoration:none;">Underactuated Robotics</a
...
-            Explain the trade offs in the design of $L_{1}$.
-           </li>
-           <li>
-            Explain the trade offs in the design of $L_{2}$.
-           </li>  
-         </ol>
-       </li>
-     </ol>
+          <li>
+            Would a taller or a shorter foot make it easier to walk forward?
```
[the remark this answers is not in the document]
Comment (iamsavva) [reply book/humanoids.html]: fixed.

**Review by iamsavva — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/RussTedrake/underactuated/556#-:-NvIQGzy0562peH8Gmb_:bnfp4nl)* status: 0 of 4 files reviewed, 3 unresolved discussions (waiting on @jaehunchoe827 and @RussTedrake)

**Comment by iamsavva** [comment]
CI checks are failing. CI appears to be running an outdated version of `htmlbook` repo, specifically an outdated version of `install_html_meta_data.py` script which you updated 2 weeks ago. Checks are failing because the script is trying to add references that shouldn't be added.

**Pull request merged**